From 1eb71326c942da98b4d8e5a3b2619d9aae8b546e Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 01 Jun 2021 10:34:18 +0300
Subject: [PATCH] [SW-0] gaudi: security: add fake otp and bulk sign

Add fake OTP binary using 5 different keys for development purposes,
along with bulk itb files signing script.

Change-Id: I341cd820e9724dc738d397dea8840e352e8ed0bc
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/board/habanalabs/gaudi/create-otp-image.sh b/board/habanalabs/gaudi/create-otp-image.sh
index 2c30859..158fccd 100755
--- a/board/habanalabs/gaudi/create-otp-image.sh
+++ b/board/habanalabs/gaudi/create-otp-image.sh
@@ -15,13 +15,29 @@
 
 BOARD_BASE=${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}
 MKIMAGE_FILE=${BR2_EXTERNAL}/output/${HL_DEV}/host/bin/mkimage
-KEY_DIR=${HABANA_SCRIPTS_DIR}/embedded/security/keys
 IMAGE_DIRETORY=${BR2_EXTERNAL}/output/${HL_DEV}/images
+
+echo
+if [ -z $HL_FAKE_SEC ]; then
+	echo "*** Creating EDSS keys OTP ***"
+	KEY_DIR=${HABANA_SCRIPTS_DIR}/embedded/security/keys
+	OTP_ITB_FILE=${IMAGE_DIRETORY}/${HL_DEV}-otp.itb
+else
+	echo "*** Creating development keys OTP ***"
+	KEY_DIR=${BR2_EXTERNAL}/board/habanalabs/${HL_DEV}/dev_keys
+	OTP_ITB_FILE=${IMAGE_DIRETORY}/${HL_DEV}-dev-otp.itb
+	echo
+	echo "FAKE OTP FAKE OTP FAKE OTP FAKE OTP FAKE OTP"
+	echo "Do not write development keys OTP data into"
+	echo "a card with no fake secutity running"
+	echo "FAKE OTP FAKE OTP FAKE OTP FAKE OTP FAKE OTP"
+fi
+echo
+
 OTP_IMAGE_NAME=otp.bin
 OTP_IMAGE=${IMAGE_DIRETORY}/${OTP_IMAGE_NAME}
 OTP_ITS_TEMPLATE=${BR2_EXTERNAL}/board/habanalabs/common/otp.its
 OTP_ITS_FILE=${BR2_EXTERNAL}/board/habanalabs/common/${HL_DEV}-otp.its
-OTP_ITB_FILE=${IMAGE_DIRETORY}/${HL_DEV}-otp.itb
 
 source ${BR2_EXTERNAL}/board/habanalabs/common/image-utility.sh
 
@@ -52,3 +68,6 @@
 ITS_SED_PARAMS+="s#@OTP_BIN@#${OTP_IMAGE}#g;"
 sed ${ITS_SED_PARAMS} ${OTP_ITS_TEMPLATE} > ${OTP_ITS_FILE}
 $MKIMAGE_FILE -D "-W no-unit_address_vs_reg" -f ${OTP_ITS_FILE} ${OTP_ITB_FILE}
+
+echo
+echo "*** Created: $OTP_ITB_FILE ***"
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK0.crt b/board/habanalabs/gaudi/dev_keys/key-GPK0.crt
new file mode 100644
index 0000000..c3ce2ea
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK0.crt
@@ -0,0 +1,22 @@
+-----BEGIN CERTIFICATE-----
+MIIDtzCCAh+gAwIBAgIBADANBgkqhkiG9w0BAQsFADAfMR0wGwYDVQQxExRIYWJh
+bmEgRW1iZWRkZWQgVGVhbTAeFw0yMDAxMDkxNDI3NTBaFw0zMDAxMDYxNDI3NTBa
+MB8xHTAbBgNVBDETFEhhYmFuYSBFbWJlZGRlZCBUZWFtMIIBojANBgkqhkiG9w0B
+AQEFAAOCAY8AMIIBigKCAYEAxdld9NV7ZJ6ayyEEzal9rKJIRLwmYqcI5K8/AqWN
+OfHkfvABb9p/cndynX2HJUgRZLGyVB/elCw/25inIN9fDnkFoS7MS9NUlWP8gMUo
+xxncuLqKqRolTGHXl/8/+qqD57wZ+5DRhYh5c0icVOGzpd5L2wvRnN8FJPuNSMG1
+EgDel3H9rKlZwz8sYBk7jz6XOzhAHeJrpSoeCjwT5beDFFWg32hUUm8UIdLQKvBN
+Ekm1vFBBiGmu/7zKOxcTs34GgLVl2XwCAUibwQQgXDN3gKe5cbOcYMPgSWEt3I/1
+AVPdKH7WmYYrGE5xyHpE2J1/bdf59p/ODCMOZfy1kxwWr7KmchAJV95pGpF8L7V2
+RV7FaGgYSmumAvvN9eRPUPWODU3V/gucwqAKamhV3atNPd/35cYf+K3Gtj1ogRkM
+8t6846ofwUF9+Wx+bP5iXkqmN0qgXKH73tCW9os/9zkfCLNBikee0NuQVBwEO3Cy
+3132sK2gKc9Ddq0C6NPT434XAgMBAAEwDQYJKoZIhvcNAQELBQADggGBADP1H1I6
+4IIYMJAQKVvyinYq7mTps/VIegevLIvIosr/gHMVVdDaLG8tsXdVUqc2Gw3B886d
+Jhlr3TmJlbX2mkb+DHNNPP4gxsann86mDE4j9SsxaZmu9HYQ1APzPX9rtrU1ZjSL
+YEpsnJRioxBsc6owgT6pO0MJhjCpSFaggbTxvRSaGJUyltLI6kg8QlGMNcMfU1Pc
+/sUNIXPd6xLgM+1qwPOHRrifdMC6/U8H1Xmj/0wmCywY90+zgQCCQE6VVC9VaJ0e
+I8C18EJ7WYx688SFyvefl8H2ia740uWh/ZoaALwwfcJv+32+puHmMWMiqGv0zIqt
+hKzBYlsc2lo6NtngPlw2kLvQn8p0Dgk46xGgT6NswRQHp68y0KeLjun2azMPj9ct
+x7A6jGI4WVhJ6tKP++PYmAXm5SzfW3o7NqxVOzlSrAJKfhnmVDZXsun4m4QwtqOv
+y2LoY36dtegfLpchOdazgZyKsz5uu6Rj7ipc3r99hYJpU631DxCGHAIzbw==
+-----END CERTIFICATE-----
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK0.dtb b/board/habanalabs/gaudi/dev_keys/key-GPK0.dtb
new file mode 100644
index 0000000..9bb736b
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK0.dtb
Binary files differ
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK0.key b/board/habanalabs/gaudi/dev_keys/key-GPK0.key
new file mode 100644
index 0000000..0b0b5e8
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK0.key
@@ -0,0 +1,40 @@
+-----BEGIN PRIVATE KEY-----
+MIIG/AIBADANBgkqhkiG9w0BAQEFAASCBuYwggbiAgEAAoIBgQDF2V301XtknprL
+IQTNqX2sokhEvCZipwjkrz8CpY058eR+8AFv2n9yd3KdfYclSBFksbJUH96ULD/b
+mKcg318OeQWhLsxL01SVY/yAxSjHGdy4uoqpGiVMYdeX/z/6qoPnvBn7kNGFiHlz
+SJxU4bOl3kvbC9Gc3wUk+41IwbUSAN6Xcf2sqVnDPyxgGTuPPpc7OEAd4mulKh4K
+PBPlt4MUVaDfaFRSbxQh0tAq8E0SSbW8UEGIaa7/vMo7FxOzfgaAtWXZfAIBSJvB
+BCBcM3eAp7lxs5xgw+BJYS3cj/UBU90oftaZhisYTnHIekTYnX9t1/n2n84MIw5l
+/LWTHBavsqZyEAlX3mkakXwvtXZFXsVoaBhKa6YC+8315E9Q9Y4NTdX+C5zCoApq
+aFXdq0093/flxh/4rca2PWiBGQzy3rzjqh/BQX35bH5s/mJeSqY3SqBcofve0Jb2
+iz/3OR8Is0GKR57Q25BUHAQ7cLLfXfawraApz0N2rQLo09PjfhcCAwEAAQKCAYA4
+CouKE8RJaoGXIwmsxaMi+K62xYKr8/tZZ2hWNBWOQb3C1PHbQI1UG5B8GrfisAi5
+OHo2SPavit8lsRY6Aft3aoFVOB/HJJuzWyiFQxN7XhB+xNwM7Sp+ecJrO/Jf6tfF
+FqtXbhxjEhawuHMLugi8ZXygFbdwk11uLdDFuJS0/kkOpcvAkhGqc+T6bXtI5mZI
+8HR84osyM/1hUcYjDfmTk3OnhHJKDUurDv9HAl0aXoWbSzJc1O6sYsn3teAvOfVW
+Lt+pNSZ88JZysUCIgREu0CY6aGzaptgSfrxCw2Suequ/I3yGLhzfslZ1dx9dKmj2
+/8TX4z+uosjtAO4Z45FePrDbdiNPG2NaWLxAtaxYaXIHDU/LlIZSsZ1G9rfKkM59
+I1TqhDzH8/stFZOx4o1TXIJkB1K8YNzvBE9J6+hRw9YJ0sLy8OTDz8/qE1a4TxWb
+zjcZeSAygOzVcnAHT8K5NVFH8OhMYDTr416Kt4kl7am0qZdui9q+Aczp86tMUaUC
+gcEA/XbqSTZSfUYgDcK+GXPY9CTbgSlKcUzkebNlyWrJt6JrTP7ImNJLQhcEwE5C
+hLMGCpq25D9Dx3e2axnPgtkxHgBl16ekHsWDXvPsJQAd2R+A0YOWG6908EiPSfXM
+6mzcCPgRplfs0kmxhXUDH6TKOyhJy7lB63goR+k34+Tiz/CpvPX2TiOrtDEWinmh
+kCffM0qPryKsYW/U/RWHZRysCHZosh76D3oMRPw0ZoqgWxPJZz1ZVKSniyttmiVP
+oesjAoHBAMfUB27ABn26IgRHW+EMXifgl/HzpDObmeooxMPciQfTKM/PPpISxeD4
+Xd86Dpx1AwLf/DbKJ1m93QwpMb+QWPo/iN7Ecz6eHTyFX2yWZhTG4K8lOr4Kp3Kd
+Li8ZoPaYWMsLe9IHR/u+zTdhNYADVkhFrWspvbXIf0lqTHtDO/qeQhI4gW/rVN0Z
+Uce74i0bWjQDqnBxkqQNPqKt6ujXrrGMaxwaGubQrxPGSOzDGfpYkMDW7Myq5CgR
+DAM0g/d6fQKBwCFcb4XnYoAepMVq5n6PIvUvjLy5ZJT5G/FwzWrvZVGITTs7JcgH
+leRDB1sQcezpa1gl+QCd2xv8A8ShBxEf6TkbsEvsQsvSD3KGBITyOPuFC/2HplBK
+jce6pTGE6s3IOkXr8NbGJ+Od04HPXvPhSsTyjpQD9Es764FbjwGt9LSN5ql6bkrm
+owj6lQ7w53UD6nrjy3mvxKSmhZAyLzn9xh7vBuS68wrqc4b48yS0Bls2hYHTL1jW
+uz60Hp71Xj/fpQKBwEJr3+yYN5AjBpuVfJzZl5J5vQfXjIT6+7YxxvWzoHjRX4dz
+w8CaLNwSsmE40xJ+cnnKKfOsTlguHAOmzbp3KJxF8iDZ+9ScietMCeP2Ogg98wFE
+xrnUoMQgKvHM+wv8p8cWEhpGAU9UpLH7xeS/Nzbesj4kqcJ62pR0mOd/+bfcUunr
+jlq9RoAXom9ZrFyB8wHHFig6GBwfquo4FRVAW/j4S8ITi02doBRanHLWhxQHzFXN
+GvkM9Lm8sNl/4z2noQKBwBsiHbut7Z53fFxVk7/J4gbq0JooXg4QyoKZCQfLkT6w
+1bvs8TSAIE8tplOnzxe10fZwBNUR3gJj88Os2pwfpC52hQn3KeQAhL/9YSqNtcTN
+xTYo3Z7xzyv0NM0Sqii/1MQIPW42j4yTOjsbllUS4jj4jHV8YAgVAeKm5PebtC+2
+uAZc/BQJZaZG/wTs22f886wqOC86funWPXDuLJ3SASn9V+kIee+m4ZDr8rdHvNxQ
+7KHpssorVYrLiSCTKiU2ug==
+-----END PRIVATE KEY-----
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK1.crt b/board/habanalabs/gaudi/dev_keys/key-GPK1.crt
new file mode 100644
index 0000000..10cd2a8
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK1.crt
@@ -0,0 +1,26 @@
+-----BEGIN CERTIFICATE-----
+MIIEazCCAtOgAwIBAgIUUETe5KFUd6cim6g0eZwA1hD4BngwDQYJKoZIhvcNAQEL
+BQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM
+GEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMTA1MzAxMjAxMzJaFw0yMTA2
+MjkxMjAxMzJaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw
+HwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggGiMA0GCSqGSIb3DQEB
+AQUAA4IBjwAwggGKAoIBgQDijfJZwSa8Xo190sWxUWamIMX0lmjTohHt1SAzLFmp
+Ve9o0eZoCrbu0ZJA/RTspZ8h+L6HJWN4Zn99fr37FxaewzeB/OT1P6hZab5xcxoC
+ugN/K7LTYImGXRJOW/YbHGcBfHQS69LLYzOi48ItNxhLFbm54Dz1VNgwL64sPHY3
+9oX182HcVJyQNYxxTe68kIOMTs07VH7pezweWD1sfpEBKYA+EzuvwCOvap6IwNoV
+tNION3uf/UnLdbxRNOCxlTW7OO7olLI/dfv/adYEnyg5qGzzfSLs7WF1z74O8jbR
+Xnwo2NSWmk1AZd06YtvIeUA0Dup64l/vyAncvHGF1GUdzPnadeFwwdhKfZ7e8muB
+cenP7zzcwgvYUTJmlko4bVNVAz78uWRCBJdJf79ybnoNFIRN/xUZFac9CkewmawT
+oKkTNHVz4lLJc6KiQKBXBF3FWlDC+FQrHuP30Jc2wEDBMwutX+6xg7YgqNWs/4GL
+6YxqbVMmYga4SKiRG7pFJzsCAwEAAaNTMFEwHQYDVR0OBBYEFK+y/6sk4HmpAfeb
+fja2QabUkFHcMB8GA1UdIwQYMBaAFK+y/6sk4HmpAfebfja2QabUkFHcMA8GA1Ud
+EwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggGBAKZWkT8kR0v9UTh77gtfwKsM
+mBUD5oo2/bWDy4XU/rDP5kF3cbZQfQ5BUkHceTgRWleAD82gvX04fwRKiT7DoDF+
+dHGV5PIWzzUq06iGldw92QqKJ2EgTHDDzTFiZBOcg0Amc5CvjJIty7UkRZdHBjQD
+l3QEHE1NMoh++ukUZOf/RLoin6LGOZuR7D/Y7kXrrlHOUN1ws0PUH/OdbyFRbKUp
+NSXZs0VofkYdk26kCFqXabyENt6QoBvizQP+6XE9iaFYj7t9luXa6JG2egY6xmQ+
+YySyvXzc25BZYWGy3Yv6pS0lIgpVORdKZGrBHLKjL+H72XJN9yJnEs9BxnUNsDgR
+vKajVwPikLcVoLujnXA0bsTbCmGNktnY97H3ulQoGCUh3A1tgeXhNJ1UHBPUh2zn
+6BSkbKiLTFliEBSTE3HjvO3QHW9OGx4sKygZh4yUIeoKQsaQVbLtUi1zc7VWZFIQ
+WujXTrDfUqRFG3Y5mt1oBDxqdneSPBHyTOKLhTMn6w==
+-----END CERTIFICATE-----
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK1.dtb b/board/habanalabs/gaudi/dev_keys/key-GPK1.dtb
new file mode 100644
index 0000000..ddcb1d9
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK1.dtb
Binary files differ
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK1.key b/board/habanalabs/gaudi/dev_keys/key-GPK1.key
new file mode 100644
index 0000000..3ee6f01
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK1.key
@@ -0,0 +1,40 @@
+-----BEGIN PRIVATE KEY-----
+MIIG/gIBADANBgkqhkiG9w0BAQEFAASCBugwggbkAgEAAoIBgQDijfJZwSa8Xo19
+0sWxUWamIMX0lmjTohHt1SAzLFmpVe9o0eZoCrbu0ZJA/RTspZ8h+L6HJWN4Zn99
+fr37FxaewzeB/OT1P6hZab5xcxoCugN/K7LTYImGXRJOW/YbHGcBfHQS69LLYzOi
+48ItNxhLFbm54Dz1VNgwL64sPHY39oX182HcVJyQNYxxTe68kIOMTs07VH7pezwe
+WD1sfpEBKYA+EzuvwCOvap6IwNoVtNION3uf/UnLdbxRNOCxlTW7OO7olLI/dfv/
+adYEnyg5qGzzfSLs7WF1z74O8jbRXnwo2NSWmk1AZd06YtvIeUA0Dup64l/vyAnc
+vHGF1GUdzPnadeFwwdhKfZ7e8muBcenP7zzcwgvYUTJmlko4bVNVAz78uWRCBJdJ
+f79ybnoNFIRN/xUZFac9CkewmawToKkTNHVz4lLJc6KiQKBXBF3FWlDC+FQrHuP3
+0Jc2wEDBMwutX+6xg7YgqNWs/4GL6YxqbVMmYga4SKiRG7pFJzsCAwEAAQKCAYAT
+piQCzNHDPKAg7Yrf2iOA/Q6uK+1jjYbojbeBOLN+EEn3WV0WYPeEtwBsSobqD2NB
+2wULS57Y7mrYQ4nOoS4BDwVdbMWez6qWNqmhxU3YNrHe8bdMxe6njKLt0OQDbe8x
+Hc4jZfieUzhDRGWu7jxZJsU47c3bqUQmybvdclD+gY8UycsPBPefOkW3/f5atF30
+aVE7eBSZw2pvZ0/MfR2ucB8DQz/F+nAd4q1lRvSjNU+SlBVQxvMDhm6SzHdAb6RT
+lVHZIM3sKY5+ALD2ZSu+g7H99XMeorlLZ94/StOPb4zBOBIT46JDO+EoDyhVx32o
+K05q89di9d2DvNOMI46nE3C+vd93TEDuBI17sGmbkEBMQXEOsSKzqZ0vJyJ/nLRE
+z2+eEdEaqgMWu0m15dKSwhmMS0m3Gy3l6Jb1e+anHXE/2a5XSUQXRMCqEPwpqhXU
+AW2vaZXGu57UC/3dns0l5yUIGKapyOkghQC7/0As+pwOx5RX4pAhBwoG27s6sgEC
+gcEA9C8IEai/bwBe6bmV5KkcwjWx3ysFbugsAZze3GSWqw33rDYi3tYTTrmITbam
+ldzxEFtO/3FALZxbCnx/D+kyUOGVSd9Qer71NEsU1gayQAEKZgp+SulZBRuaZlyX
+wMuuX9FwiYMN7pDpL1V0PieoDQxCuyYaO9hjeQ/xyLDiMsN+DppGSN90CKnOnLiD
+Q+dbFyD7z+wpjTwIjkE52HskNSA1HyTeuTzGh+H/Po7qqxlReoSJ0Sf2g0B4jLnO
+rFBBAoHBAO2Ehc5VvfxCsL3PSKId8Vu6xRfy8eQVyUPdZ26tNFCw5EtqhIAzpS/p
+Br11PinKdEREJ7iXwIKzI08r+2zFa5xdFB4SadCNPFdYwSa9ucwTjwqxhXRh6eqd
+gURPPciOjaZMz3aSEr8LSp+NigZbsnJmdpyJzS1eMBRlINjT4mx6yX75v22vqqoN
+B2s+sK6qdHG5tMD/UmqoxuPCFQ218aN7A/0hWyoy/N0jvbSFTKTgZcBdYtlUXYMT
+lHVCqNSYewKBwQCradCqROGN1J7MSCF3UVgE5Zil5xd7UbSCtVKlOF+kWLvCwoYc
+E/+dxhRvKYcrsNvw9YgxPZkl65rlc5vm2Xce3BzmrMNcR2zudr57uFutHo3kL/2U
+LNb3BPeQX/OPvev9C5OKlNtU25k2RzDA6SnJbw/xPMAxvqeltpGupq1DoYR+T8KD
+KYnVAw8taUGMi36gb+EVht1sbfY7loI6+8bhHGjuIf3P32Amx3lcuwr1bWKLNTth
+/kHIxZtafp0czYECgcAQRcYNDt7HS6JkPrFROg/xIuih7WkZIUz6s06EzKUK6jOZ
+3yBY0DD/CU4G6ONrZY11p924zdzIl5FRuUeuehAXVGEudkB9o0+BBHmnN9Rk4Oop
+52OnR5YJFuVmUq9+8BVQUW7dRsC2CgYAzGeqa9nb8J0ruF0wQIW2v/7yHAIlQHxR
+kLbGwwRERkQ92rusG82ChSCXIGb7G7kUnZVnrRFzC9iIHSLrzzZ3yB0VIxTPLviJ
+p1xAW4FUm6DOL9T1S8UCgcEAw86Jzi7uovOgPfCPUCYc5klUAUZqTDLOAinD4Ivq
+j0qREI6rOVOZ2SZFwZY/phXYAWbDHxvcjPbHXtSU6oaqj0mvDDLvDRXHwbr4WTQi
+8nNSXdZgTw2C+RfV1YpzwvwBLqAg9iJ64nl9zZQbFQTFESqjkQbfqcB3ZgyGKKO7
+ykp8GEBLTOIja9HoGz8DlMxf1WX6mu84ddtdVfp2aaDjZg+GKfKqmW/sg3nX7e1Q
+07DFJ0o9lzCosnmpXTJW4WLF
+-----END PRIVATE KEY-----
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK2.crt b/board/habanalabs/gaudi/dev_keys/key-GPK2.crt
new file mode 100644
index 0000000..79603ad
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK2.crt
@@ -0,0 +1,26 @@
+-----BEGIN CERTIFICATE-----
+MIIEazCCAtOgAwIBAgIUGuTxOVMvRlWvqOOTlvSb0MRkQLQwDQYJKoZIhvcNAQEL
+BQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM
+GEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMTA1MzAxMjAyMDNaFw0yMTA2
+MjkxMjAyMDNaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw
+HwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggGiMA0GCSqGSIb3DQEB
+AQUAA4IBjwAwggGKAoIBgQC2AhDLkpE9kMNmkyY+B/TYWbTw4dSyIDB4hsafaLZb
+4Vf/sTwfKbeb3FIayEZgHO1dTucY5HY/ybyS0eF+Qst6HT9LW+iaexMErvA21SVK
+yIIUz58saTRlmHBbX7aaUWmV2F9eIHmmtgWMDUEZDcM5mzXeGfRW6DnxU5q01U0V
+ZjViqd8ESHnoL55+ntqAIes1HVL8JDTmwvtHlDN1upX2rvaZXW40Tsi1jZgP5DRP
+bRdBHNUkBAg8nbVXib1eNtrXpGOk9pY1nNEOCrmKvtq1TUUHXkJ2gIxlYmjBna05
+zrVxkXqPSm8vlSZ/4zR4tY0QY5BjNXvVStkaNR+7ekxn41OZzO4xpcjAFOEn4xuG
+P91xzjsozpko3+dcxQhT2zePRWGPFZtvs9lBBW7D0gRLXEUFYZOPpAQyoulKmEe2
+P0J9VI7yf9sa/bnDH/ClNVmXz2sJmj2EElJKha4iebFou/20JuhORz8qE3HhFXX8
+8zeaM16Ubs2fVh+DJHxLZH8CAwEAAaNTMFEwHQYDVR0OBBYEFGQsp+wft7j82/Qe
+lE3xDAr3iHk/MB8GA1UdIwQYMBaAFGQsp+wft7j82/QelE3xDAr3iHk/MA8GA1Ud
+EwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggGBAK9nu3pyq7HPSB4qExN8VNB0
+JSGhTTp8iyIiYhhniuhjJVMdQP1kFweCuV0rVNw7m/I3Bf+et+rLQxWiR/HZ6cpJ
+uJ6vjBM49yL4Y2XTj7OcZt9NSl+136p+xpJgqFIV/bzDyny9mBI6RxnPM8ECArU1
+5fv1d2Zv69Xirc2Hqu+ZiNkcPXo17ACK0lU6X8bFQIkDL5naMySu4GTu2htth4pg
+eqFYcmk98zVjxUFdabMb5h/TabzdG+un+tgLXKdbP9NWvuCbwiI1KXTY+f7AHimS
+1eOcHqlIYnfye7vm3R+6tVsKXccIUE+1B3fQeeY//WBsjDHkInPV4M4PWA9GP3Az
+Vt+Ih4lbsG5b0nmGcm2QmL0GMMXpgpsg5Rs/DFyXcoRcHDPQ+erT34gfO9kCV774
+51ZfMOcCqyOmUZZ+RYC4PvyrbPoYFpKlBpFh05vf9FqElWNkK5ORtdSu8V5LYyfJ
+7IbDC+YmvptUhJEHAtNYFsWoyIJtqc7AdIs+/sK//w==
+-----END CERTIFICATE-----
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK2.dtb b/board/habanalabs/gaudi/dev_keys/key-GPK2.dtb
new file mode 100644
index 0000000..958bb9e
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK2.dtb
Binary files differ
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK2.key b/board/habanalabs/gaudi/dev_keys/key-GPK2.key
new file mode 100644
index 0000000..b1a015a
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK2.key
@@ -0,0 +1,40 @@
+-----BEGIN PRIVATE KEY-----
+MIIG/QIBADANBgkqhkiG9w0BAQEFAASCBucwggbjAgEAAoIBgQC2AhDLkpE9kMNm
+kyY+B/TYWbTw4dSyIDB4hsafaLZb4Vf/sTwfKbeb3FIayEZgHO1dTucY5HY/ybyS
+0eF+Qst6HT9LW+iaexMErvA21SVKyIIUz58saTRlmHBbX7aaUWmV2F9eIHmmtgWM
+DUEZDcM5mzXeGfRW6DnxU5q01U0VZjViqd8ESHnoL55+ntqAIes1HVL8JDTmwvtH
+lDN1upX2rvaZXW40Tsi1jZgP5DRPbRdBHNUkBAg8nbVXib1eNtrXpGOk9pY1nNEO
+CrmKvtq1TUUHXkJ2gIxlYmjBna05zrVxkXqPSm8vlSZ/4zR4tY0QY5BjNXvVStka
+NR+7ekxn41OZzO4xpcjAFOEn4xuGP91xzjsozpko3+dcxQhT2zePRWGPFZtvs9lB
+BW7D0gRLXEUFYZOPpAQyoulKmEe2P0J9VI7yf9sa/bnDH/ClNVmXz2sJmj2EElJK
+ha4iebFou/20JuhORz8qE3HhFXX88zeaM16Ubs2fVh+DJHxLZH8CAwEAAQKCAYAt
+qCb+ai30+7OIr7EQLn7FJvu00XHBnUGLeoOoyWO1eLnScmtaheSBPKt7/HFq+taw
+aXKfU/A14C1lxzCBs8/HOj7+cHfRCL6klh28I4vPTq0S+2I6icmTEsJcqY3s+C88
+X6LMwejCYTbTAyApYMJwWPcSWABrLKYcW5SbeEZ178XEkkhXkrbB//AKY178dYXD
+NS/0AEElAhR2yD6AceXck9dYcY+Ln5fJs1NJ6T0WD4+dzdXJyxkl24n6A5t1Xtzq
+npSf3DjbbMJTx8OI+hKHxFLA20GwVaxfBU055q8Uyg1xKLdQs1Ywk0LeKkdIsBdi
+xoh8hvTlzFTrt7CSvu69eQcQnkm6OXcny2KD2+J35A5SsXhw5tNzHEhgRRqEYWJ2
+Wx7ExE6+C9VNlxOXfFsrHypEBGFBYymMAJM2VF6ZJQkBUsYR3NmmrmGiBOY6IysP
+Ymumf3y6jRuLpryTYUpsFrg/Y71933dnZ1biRR7i9BO9WrHJ0PatJCigf+JNIEEC
+gcEA2WMHQjqpts2Qa6eAXhG0HXT6mHEf0sAHPwo32cof0ig9R90+eYUNNC38CmqT
+Kb+kRgEiFGziec9bGwcM6En3+Op3sGziVgFRQ3UW6yQ3oLNEvWGbWSe/AGvxmpj3
+z9K/wT6U5RUNabtVPBOjQ6EYF0AWGv+fUvkeKlNWFWZxPWUFsiLLbzyoWz0URevu
+wPp2mk4FEprj2k9Ltg1R9g1r1+UyaGiO2SpmN5FRs4DG7uXVHCK+NZ4iTPdalKPi
+IS+NAoHBANZWTRZbXhzOij/RG8Nyy6AVqqYE46x7bsckzOKIixvEVRuIAOPmaYUo
+0zwWQZbZgVltl36BEj6vxWCGdFQhe3svGkw0KOAztKIU2pF7Jdy6N+4a+rafrwZy
+yNdDsLHwsTzPiIYTrz9746uOfaoMgXObSiXTW+rgyyMvklpTBblzLl5h29tqYmft
+XRrv3IjFVQ97vsWeAe5fpSavSzLPN8dif7Vx6NE4a7abkMSggBbZ1NRLcFbMrNid
+9pBw3LvrOwKBwQDFOQTi31oOEV3Pbb4WmjUw7GpWV3uptrJcTSbynmBUSFGgm2cM
+1Ntv1bBuutYc8FNXd7u1kDvAi5F6aeEZ9dHVsNM56u9FXBif/OD4IHmfgEUDFETe
+lCLDfStLOTDLCccSyXZIh9sNa/oecSqUNtkmbuwfPn8LxhvKBqV48OIX80lBnjO3
+De8oJ7f23+BSkGl+AP61LHVAIOTtnBQaCX8n9jbPZt0CxWbLTzoygsj5AOdhiVvM
+RVWVQmyZoj2MIhkCgcATwPBp1xHdZcuoBgWhBbtujZCkrF6vTtDu2Z1oY10uufF+
+EBLE7j9bXl/ZcdOpq3IYY9lKd2rPDDQMqVefSVawC69IxqXdw+Yp6fIa5lae+Sqe
+by4cFqty1LFd5YcrXxt2l0nl0XqdRzW8Ye2C9pQXq4Ax7EDUbNNL8t5yJKSqassj
+N7GjjzJykZW7HdAdatJkEuuKGv3ikj4OSqmLjT1+yr0qY80pOyHyOnQlKwWjFYBf
+8HPyC/sCG8zVkxFE4HECgcBpev7eqVVGz6Iq+0sOp2y7bcLYywjwKxrRLY3zhMBX
+i1in5IfSi+5YHBkwNpP85Kb0nhN4k6E8nspO4hN/cO2qFmMSn0mPRxcESKMWiZCA
+BmG3e5VyeZuetsn/HuopLRxcViMO3OebyPfWhJ4CQC323uilWKdDlw1JAd4aH/jQ
+Bqs1ooAdOnfa6kUhaacy6ePO46kVrnvWaENLA1FGUQryq2+2TFd7RQ5Pj7qJfmVr
+xG5NR/vHjhv44ESwDcNzlqk=
+-----END PRIVATE KEY-----
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK3.crt b/board/habanalabs/gaudi/dev_keys/key-GPK3.crt
new file mode 100644
index 0000000..5302722
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK3.crt
@@ -0,0 +1,26 @@
+-----BEGIN CERTIFICATE-----
+MIIEazCCAtOgAwIBAgIUb9cDzpk0LSEKBrci4G2t42G2lQEwDQYJKoZIhvcNAQEL
+BQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM
+GEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMTA1MzAxMjAyMDdaFw0yMTA2
+MjkxMjAyMDdaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw
+HwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggGiMA0GCSqGSIb3DQEB
+AQUAA4IBjwAwggGKAoIBgQDMyMY4UW34HKHSetRbNPYnZ086ieFffwWeG1yEh15Y
+AsPbBmbd6c2v9tPSysjYprdq6jq/iVb/knhpKv4r9C8G0JX23vpVML0a3neIF8oR
+ceh/YiEEGVJ9kAeOWXhT9+INilNj9tdlotymhKEQIAGFGXeL+t19Rmz/GRnCGuHM
+y5wSPAoJmsdVu5VukrMopgAV/ahapJBtOahrCaWnaQbez7/w1eJgFNC69pMwBggR
+qmq+zA6ZJneh3zct5EDpJt1oCObKv3jOdabmpeFBXoq2LeJl17kuUrazt3LALqe9
+boLLKZ3y1+hQjo1+ErDWOeLhfV1BqReBfiyL/lNHezqIZSs+OAGZwMMPLrwSJJZX
+jnBomVXl9EJFxbX54X/5SZ1lQxemMNx9BU5nn04UjjswXW7bODFT04nAC/FSPgqo
+q5eDVr1KJsEEjIdhYbb5Ra+d/bvSBFcT0/IPCZ25jgoG5dYkdaPv+4uU9cD7+H79
+uGl7RUMmnA+fz+20dKzhUM8CAwEAAaNTMFEwHQYDVR0OBBYEFMjJliPXLia/C2Ma
+o+swEjnFyNblMB8GA1UdIwQYMBaAFMjJliPXLia/C2Mao+swEjnFyNblMA8GA1Ud
+EwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggGBACnNque4vx2+6cJKSSJYnBhD
+B44J5MddI/T7RczCqaccNXVXFlrkOG6lDO6rU3Y6ihTib0da/X/HcAQ/J296HDzw
+jfp4YOoF5zXzxyRyyMyHYZeeB0j+HfZUGE8kIucqKSEpiYpNZ9ZhD55rOVvs2xra
+YThLAFnF6X+rUx8/LSNhEtrsERJ7n3P8+1+g0/1CR3hqd8VxuXybCUNAXINW2wSG
+5j7S7a9XNG1LCs6OWmSQ0Lcr/pZLwLXUdWWZK1KZspdcPacNTh0vnHcxszT5nhT3
+coYCouO77S1c/HN0D8QAlybqW0NLwN7hFVmfEzoIMV0oo69n9S6sP7jT0UUJ8XNA
+aCLPHDO6sR1B39b1dqk8yg1Ppo7Fu/hwi+QDFGm2bNF9llGC7ndek0aZN468//A+
+ZdIXWj8gV6ExZJuBxCZ2crCP1mrb5IO2BJOrzEOaLF0CMU+jBTC0ZB55tLz5ViPK
+hNqW+VQp15FwJdPrXhmj2cFbaSMQ3nHAoMKF94N1Fg==
+-----END CERTIFICATE-----
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK3.dtb b/board/habanalabs/gaudi/dev_keys/key-GPK3.dtb
new file mode 100644
index 0000000..0ea2a24
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK3.dtb
Binary files differ
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK3.key b/board/habanalabs/gaudi/dev_keys/key-GPK3.key
new file mode 100644
index 0000000..884ca39
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK3.key
@@ -0,0 +1,40 @@
+-----BEGIN PRIVATE KEY-----
+MIIG/AIBADANBgkqhkiG9w0BAQEFAASCBuYwggbiAgEAAoIBgQDMyMY4UW34HKHS
+etRbNPYnZ086ieFffwWeG1yEh15YAsPbBmbd6c2v9tPSysjYprdq6jq/iVb/knhp
+Kv4r9C8G0JX23vpVML0a3neIF8oRceh/YiEEGVJ9kAeOWXhT9+INilNj9tdlotym
+hKEQIAGFGXeL+t19Rmz/GRnCGuHMy5wSPAoJmsdVu5VukrMopgAV/ahapJBtOahr
+CaWnaQbez7/w1eJgFNC69pMwBggRqmq+zA6ZJneh3zct5EDpJt1oCObKv3jOdabm
+peFBXoq2LeJl17kuUrazt3LALqe9boLLKZ3y1+hQjo1+ErDWOeLhfV1BqReBfiyL
+/lNHezqIZSs+OAGZwMMPLrwSJJZXjnBomVXl9EJFxbX54X/5SZ1lQxemMNx9BU5n
+n04UjjswXW7bODFT04nAC/FSPgqoq5eDVr1KJsEEjIdhYbb5Ra+d/bvSBFcT0/IP
+CZ25jgoG5dYkdaPv+4uU9cD7+H79uGl7RUMmnA+fz+20dKzhUM8CAwEAAQKCAYBZ
+oeNI0TNt7rJRmxuJlEPNeWVvPNxdXQ6LZnR3OXzszsEFA+dQaD50L3gFgmaolUgg
+zsbKAQYINQbpGT1I5ccTl4acGHq/gOa0dP0iny79tbrKzpue3KqDdOApyTVmo0/Z
+F1tqnKcowEENzW1lzmx74mq15Sa+521btpPUpsmARvpX7NmD3LA08Tb1v9YVC5ts
+BKuEiRUCCcOSnhVYkhrEuW/MrQE5hsLZooVH9svfV5UDT/Nvyyt/uWDAgHB83Fox
+F6LT4PY+t9ZCRTWtmFjW3M6jIL4seeKX9cVEo/XKSlfpxT2oleLmVS71eqW96Ps1
+8j6LUh43OfhgXuhx6aNiPk1QGqePtEXj66BoR+yAr0LLQBYuicKtL8HZpvj+StbC
+ACCwTDseC4kF8tMNNjc+nNpWnRGqFyDn6besQ//XDgUaqmzvhe0f7L8W6Y0SSzOg
+5XE+WOsgbLkUqdR7kjBCUqj8OBsS4D5PjLzVKYS8+BPTKi6mWuj08qLIJ/4N/3kC
+gcEA+2+Ur1U+z0KjNCCMpb5fQptsdNwWJo8iOZWn6/Ybv6NNgO7+mL5YkV69neci
+alqLsI4RUiBfeUVTb6BSHvXF82DHcPUJSMq9FyWtaCNzQvF8CEfl5tT79Z8JgaUL
+2KBvKvl7I3DflE0ymNsIJhGVh26PYlNZ8LXX4EDZIBiz0eRc+ujpGuMxEKaHp7OQ
+u2xfkOaPJMMK2P/vYSmGoIoBLzKzsJkYgOckA8JMbRDek24QGjiBaUI38/7dYUUK
+xy9dAoHBANCAZ3fCRlns3i181/8bbCcDAynh4MJiWwgCxwzc4RNYZrjNb53AdrGY
+DomrEWbS2F2JFGSwdsOU9i8SCmWSwBNh3omEDPSaQ9T78bkA7L4Z0oDLp64WhtZ/
+Oz/e1GBP4+faTG1bzCks2Yeju/5qL+b+TcsFzlqoSvA3w2Lit3km/MTjUJoVMZF6
+3CDgtPtZjMXRkiaHbCZWu35lzhm2/ZofCcuSJ/hhdQ3SmBHeetj3ETsn/wBwxpT7
+gJahdrF6GwKBwGtxSeMGlVlxy1XKFu/4grJuwhDJv3lorqJWzjPprf/dI5BWEFtq
+82PjTTm9fBpgaNLgUox6ovGqazYHFzB9q8INeAxP2VDOubC7e/dFNsrDcFIFjztt
+heF2vVQ2F/LQ09BUXYdOg2vW9SkWZ8wEtZXy67iDj6ppuj7s1xt9EtI9OHJLttiz
+ftLzpoJGHDHK5KeBrmdyJR+Hdl5ZpVrSviDSFLEBKtRicSReEdSFYec0IBNkVACL
+uGc+YJFhNHLbTQKBwGsrdQNzkmGqhNDwuL1hxQeQXtHqTnLbx1qowtNzvSUKbVJU
+FikJwniqZd6RVZJ6kcgSK69NnN4O9TNgnIN1GNFdntskStx3ZqJVFfw343gbIb4d
+8ARlg81A/J1p1dPjfIuyA1K6uof3i6BSotC0mAatdCMEyvqCQnKS1xU+33ugdJBn
+Hwj0F9RlSDyGA8CmWNtdSzgOp6JVNm5Tucox6syLY6eFapIa1eXS92NH7fETnYHA
+nywEy63MUF4chDvZPQKBwDydi6Ioxa2ejF4qXdhM9W+Fi0T7R+eGRDocgDAXR5xD
+iXe6WKUexL893ol69hPjzkKe90/iROo3/aZofnSdO3UH2wNpup5noFK/56ko+9Bb
+SPiB40iImrpdXDuqGbxIcLSRIv/PJeSHX1LdkkfQab3JrtkgUdYMa+5SGv7yUDKu
+qMNyPgMHzZmMGEn7759iOFt2D+5hIAwV9rYuTB3Wq1pYWGoJbkt3L4Uq8MFhOWuQ
+OWoYRgAVCEawrqtLEt66Rw==
+-----END PRIVATE KEY-----
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK4.crt b/board/habanalabs/gaudi/dev_keys/key-GPK4.crt
new file mode 100644
index 0000000..2caa1e5
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK4.crt
@@ -0,0 +1,26 @@
+-----BEGIN CERTIFICATE-----
+MIIEazCCAtOgAwIBAgIUWRj/45F1jSmTt60eEhxhQK7AUGowDQYJKoZIhvcNAQEL
+BQAwRTELMAkGA1UEBhMCQVUxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoM
+GEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0yMTA1MzAxMjAyMTFaFw0yMTA2
+MjkxMjAyMTFaMEUxCzAJBgNVBAYTAkFVMRMwEQYDVQQIDApTb21lLVN0YXRlMSEw
+HwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggGiMA0GCSqGSIb3DQEB
+AQUAA4IBjwAwggGKAoIBgQDUd427G/bcJRsiTKy8GYMSQIRh67/a6drL4twE4tWz
+yT2qwRyg2NeyBcjMBrpladFeEUPNcnB4avgrY2B3baBeioLXIOtJv82a4H7cv+on
+OQjebMpARbLDyFblCjHJ7Rwl1s+aVpDcMiPM9g0+IybAMZ6gsTgtSqaiBD3gJ0RQ
+J3c3VmcpiJx8lRJst04+alwvGPNsAZNZ2Y88M24ZK+8RJ3ayQ8+QiWukc7eBmY4X
+XnJVXtuijegr/dSCgqYoxpWwxw/EmlOvABE7ASLebTmCcYWDEVGQvGPskaF1V+bB
+2VxKZuPTes7MWhFxB0T3c8i8OTV7GddjmKnnrzemqyBhd4pjazAMCV2Xw7Z86odD
+9202hCSWkU5WYOFMz4RCeSvK46vtBTvefIVsOocz3ppwDD/SzukRarGiG9BLizxg
+9SW+iYgAmo0KbvuSMlZT3WZ3YSpVUEUG6ktto5l+3/OUQI1/OfeBl0ibkRMc2dLP
+HmRsIIxfcjnVp56iOsx8iiECAwEAAaNTMFEwHQYDVR0OBBYEFE4UtYcDIyTPv6oR
+zqFRmb8kZ7HJMB8GA1UdIwQYMBaAFE4UtYcDIyTPv6oRzqFRmb8kZ7HJMA8GA1Ud
+EwEB/wQFMAMBAf8wDQYJKoZIhvcNAQELBQADggGBAHgw0dtB8g2GavYayKOtDYd+
+6XJbYCugsnWYPomKYFwfV6ibIEsSjw9wDsSLdBP2983EYNjEwfP5ZYy90JPUQHMJ
+h4JekG2dhsnNRY9lC2VOYtD6UIhdY6ve3n4Cyoag+oVfYI2Lo30VJMPlkEXoV9wu
+gNN/01MtPEGNd2GC2Q6vZZvw8/yowPi4fpDYsRy50g0hjr3g/ce78zRHNtWnIq6g
+cwHEfWD+wZ0yLGxwtY6Pp2j76e6Q4Ve7oHUJJVecWLbuA0OM3YnIxnLsc0cTNRMI
+dGYhgaa1Yiu0sFR31cfNqSRzcBGmsfpNbvkH6oraQSUAJ6ZAa5EZ68bhm7vaQntB
+fMg7IS/hyNzYQfKcvVb8lu/VD/d97rxk+/wClZHJz6RpMb0CDY+EabcqWGphHixq
+VdGSLgyd6bP1xt8cIMMGEYkzxc9lRgaKztmxpf6G5YEcNWWfWv8yLp4ssr+n6pZE
+uE0I7kAthVSgRfMqPaVq0PIqf/txDgM2WDdRG/d5mg==
+-----END CERTIFICATE-----
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK4.dtb b/board/habanalabs/gaudi/dev_keys/key-GPK4.dtb
new file mode 100644
index 0000000..b4e5aa1
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK4.dtb
Binary files differ
diff --git a/board/habanalabs/gaudi/dev_keys/key-GPK4.key b/board/habanalabs/gaudi/dev_keys/key-GPK4.key
new file mode 100644
index 0000000..90c10a4
--- /dev/null
+++ b/board/habanalabs/gaudi/dev_keys/key-GPK4.key
@@ -0,0 +1,40 @@
+-----BEGIN PRIVATE KEY-----
+MIIG/wIBADANBgkqhkiG9w0BAQEFAASCBukwggblAgEAAoIBgQDUd427G/bcJRsi
+TKy8GYMSQIRh67/a6drL4twE4tWzyT2qwRyg2NeyBcjMBrpladFeEUPNcnB4avgr
+Y2B3baBeioLXIOtJv82a4H7cv+onOQjebMpARbLDyFblCjHJ7Rwl1s+aVpDcMiPM
+9g0+IybAMZ6gsTgtSqaiBD3gJ0RQJ3c3VmcpiJx8lRJst04+alwvGPNsAZNZ2Y88
+M24ZK+8RJ3ayQ8+QiWukc7eBmY4XXnJVXtuijegr/dSCgqYoxpWwxw/EmlOvABE7
+ASLebTmCcYWDEVGQvGPskaF1V+bB2VxKZuPTes7MWhFxB0T3c8i8OTV7GddjmKnn
+rzemqyBhd4pjazAMCV2Xw7Z86odD9202hCSWkU5WYOFMz4RCeSvK46vtBTvefIVs
+Oocz3ppwDD/SzukRarGiG9BLizxg9SW+iYgAmo0KbvuSMlZT3WZ3YSpVUEUG6ktt
+o5l+3/OUQI1/OfeBl0ibkRMc2dLPHmRsIIxfcjnVp56iOsx8iiECAwEAAQKCAYBd
+KeMy9YskcvzX6GkhvcoUecfSrC4fTsXK+zngxppQUGDC/BSSjyLdg93NYPHgwCq3
+JbjmYFTi6HzTgaruE1rrua6vqFajqv/79o6KO3jwI/k4ltmEn0tIwBMiMXstgNQD
+35J+qZCxjSbXm9ios38tbVSSP8NTUNWKW3PKU02IHBYRZMitlKKRmkhO9X3QrpCn
+cjhi1roG/vfovOB9Fx+oFPbeKg/c3/UcUkVYBb4hjx9xQbvrJnB0wcOqSqGTZae3
+A3dtHdUlvVyGFwh9udJZ8kwQrkpDBh1aXbdK55KHnIjor0YOdm9o2Oo8nhi99uj1
+67zAb5cDu7lFp6T4Qaj340CQ96tqw+KmOAMiaM3RbZIELQ0jftApcY/l+tbfCen2
+2vLm59sfgoHF8nv91KZLLkf0VFRQ/OsrGif0HhRexfwA5bKO2fzThn11+2kMw1UL
+keVBR3I8fU8Ofa6EP4lFGkRsRfbp6Wf0j7QP6siI+1CKVTkAz8ZIwI6MJPhpoIUC
+gcEA8klv25K/81G6Jk85FmeXEr9363iVrYpLo4WA6+1Ssq5TU+4GU4Ht+ow81+M1
+I6Gpoh+54xBGmeE1EbqmHyPjV3/HG1IySTmHOgRzqyXyKgXCN2GavM5e1qG/FuBp
+1FJmDUnCz4L1y40xyjmL5Za+uPzwhbuX9wy8fCRK/wVl03ocmQv8dz7d80+ey+1w
+PqvhdS/IErmRaNbgQ1po82Mrjagc++G9lPZKTrBb8/sU00gP6blUN8Wygij3ZpBl
+pBwHAoHBAOB+DF+bT61MYZ2MpPCGjkvp6JP8TalOQoG7EqSh8u6LiOltu2juRPuZ
+Z2dLqAbHoMql/11z7XV7OhBXciVlm3jEq1M2kpJP3Wcjuf3tYXmpOuDIkeqwFF5V
+cQ3Lka3LeG+QiolNaUJAJBW0GllCTqcHTMiR2sLkwfzA+64NKTAk9wlp85Tc5x6T
+dNvItPFQ5YXVHxALky7Cxm7rGWtavOJ1U7CNGs74VQ8sKfKMVesOIKXg4w/WTJRi
+gAer9vNulwKBwQCFUM1A143RN+4iuELkRhnFkh9NLOvHMqqODp6yssJnWS62pulK
+N9sXmuS7seSCd6jccXKaM1F1yIqn97kCt3+XnNKVAPR0gDgdP2faC6YlY0fLq2Q2
+caOTnCqGTvNMhg5rT2PnRNs0prt315RXv0uwhFVgcKmcoaBw+pNdob220KE9wW+o
+uantAKmS8lJp2a/UCHehfgQ52MpLpUrMaYwEhTV1j2/+sEqQ48+k/0olO6it5XEm
+wxyHLD19bQTQeS8CgcEArbv/GXKZ/BUTqpkxANAxicCRdQ2akL93Ku07JVwMJN6p
+63U3x6I2kryK0N81P2H5JzvGYG2aRQRD0GfMd/KlmpYhvPmL/119jBeTFlhCyqPU
+ZCf0/Tqd2M+o+bMkn+rK3xRukontPw9incVPauqs8gjtPH/cXF4rXs9/N5kJ2i+R
+W6C6mOdImlol4v32QpeHucZRDtrWYibs6ORbPMDxUxTX2teo+qonMeHM6OAAkjPR
+wm5Da3s+sst0J/vG12grAoHBALTmVY+1Z9Vy1vsz5WtyU3qKZ/VYD4LSuJtYxK0B
+KiI93+NN+R2ePZUJLCjdudhr7QlstpkXzGSVYxMKYS3FoRd1PqaNSoKIm5+jCW7H
+Jc0qiHrtwS7WjyPzIn7hkdEW1KZfWqhEaGfRfpIy7aCpRQJNQ+Pwu9v63Dxs95ow
+QSb2QK8R7G/w0A6PNLhCUDKn5i2apJTsGkRPP4l66e990tzoJMDH48X/tbYjwtgD
+xaN0xCwiygtHbFs/nO7yT902tw==
+-----END PRIVATE KEY-----
diff --git a/scripts/make/gaudi/Makefile b/scripts/make/gaudi/Makefile
index a3bba1c..221ff64 100644
--- a/scripts/make/gaudi/Makefile
+++ b/scripts/make/gaudi/Makefile
@@ -18,6 +18,8 @@
 	$(call hl-release-test-start)
 
 HL_FAKE_SEC ?=
+FAKE_KEY_IDX ?= 0
+FAKE_SVN ?= $(shell grep HL_SEC_FW_VERSION ${BR2_EXTERNAL}/../specs/${HL_DEV}/${HL_DEV}_security.h | tr -s [:blank:] ' ' | awk '{ print $$NF }')
 
 .PHONY: hl-gaudi
 hl-gaudi: $(OUTPUT_DIR)
@@ -139,6 +141,26 @@
 gaudi-otp-install:
 	-sshpass -p$(HL_PASSWD) $(SCP) $(IMAGES_DIR)/${HL_DEV}-otp.itb	root@$(HL_INSTALL_LAB_IP):/lib/firmware/habanalabs/$(HL_DEV)/
 
+.PHONY: gaudi-fake-sign
+gaudi-fake-sign:
+	@for i in `ls $(IMAGES_DIR)/{gaudi-fit.itb,gaudi-boot-fit.itb,gaudi-agent-*.itb}`; do \
+		KEYS_DIR=${BR2_EXTERNAL}/board/habanalabs/$(HL_DEV)/dev_keys/ \
+		$(HABANA_SCRIPTS_FOLDER)/embedded/security/fit_sign.sh hash -S $(FAKE_SVN) -K $(FAKE_KEY_IDX) $$i &>/tmp/`basename $$i`.hash 2>/tmp/`basename $$i`.log; \
+		\
+		openssl pkeyutl -sign -in /tmp/`basename $$i`.hash -inkey ${BR2_EXTERNAL}/board/habanalabs/$(HL_DEV)/dev_keys/key-GPK$(FAKE_KEY_IDX).key \
+		-pkeyopt digest:sha256 -out /tmp/`basename $$i`.sig 2>/dev/null 1>&2; \
+		\
+		KEYS_DIR=${BR2_EXTERNAL}/board/habanalabs/$(HL_DEV)/dev_keys/ \
+		$(HABANA_SCRIPTS_FOLDER)/embedded/security/fit_sign.sh sign -S $(FAKE_SVN) -K $(FAKE_KEY_IDX) -s /tmp/`basename $$i`.sig $$i &>/tmp/`basename $$i`.signed  2>>/tmp/`basename $$i`.log; \
+		\
+		mv -f /tmp/`basename $$i`.signed $$i; \
+		echo "***"; \
+		echo $$i; \
+		echo "***"; \
+		KEYS_DIR=${BR2_EXTERNAL}/board/habanalabs/$(HL_DEV)/dev_keys/ \
+		$(HABANA_SCRIPTS_FOLDER)/embedded/security/fit_sign.sh verify $$i 2>>/tmp/`basename $$i`.log; \
+	done
+
 .PHONY: gaudi-nic-en-create
 gaudi-nic-en-create:
 	$(HOST_MAKE_ENV) ${BR2_EXTERNAL}/board/habanalabs/$(HL_DEV)/create-nic-en.sh 0xFD000000 en
